flow/include/flow/genericactors.actor.h

```diff
@@ -2177,6 +2192,213 @@ class FlowSingleton {
 	static std::unordered_map<NetworkAddress, Reference<T>> instanceMap;
 };
 
+// A multi user lock with a concurrent holder limit where waiters are granted the lock according to
+// an integer priority from 0 to maxPriority, inclusive, where higher integers are given priority.
+//
+// The interface is similar to FlowMutex except that lock holders can drop the lock to release it.
+//
+// Usage:
+//   Lock lock = wait(prioritylock.lock(priorityLevel));
+//   lock.release();  // Explicit release, or
+//   // let lock and all copies of lock go out of scope to release
+class PriorityMultiLock {
+
+public:
+	// Waiting on the lock returns a Lock, which is really just a Promise<Void>
+	// Calling release() is not necessary, it exists in case the Lock holder wants to explicitly release
+	// the Lock before it goes out of scope.
+	struct Lock {
+		void release() { promise.send(Void()); }
+
+		// This is exposed in case the caller wants to use/copy it directly
+		Promise<Void> promise;
+	};
+
+	PriorityMultiLock(int concurrency, int maxPriority, std::string launchLimit)
```

Comment:
Since the launch limit vector contains the priority definitions, maxPriority (and the knobs used for it) is not needed, just initialize it to `launchLimit.size() - 1`.   Even though it is called "max" it was really just meant to be the highest priority index in the waiters vector, which is why in the original code it is just used to initialize the waiters vector to a size of `maxPriority + 1`.